src/scripts/components/widgets/modals/ModalStake.js

```diff
@@ -35,41 +40,48 @@ const Footer = styled.div`
   width: 100%;
 `
 
-type Props = {
-  videoId: String,
-  user: UserRecord,
-  onSuccess: () => void
-}
-
 class ModalStake extends Component<Props, Object> {
   onSubmit: (e: Object) => void
   constructor (props: Props) {
     super(props)
+    const selectedVideo = this.props.selectedVideo
     this.state = {
+      id: selectedVideo.id,
+      title: selectedVideo.title,
+      description: selectedVideo.description,
+      // FIXME: we are not editing duration, so we do not need to store it in the state
+      duration: selectedVideo.duration,
+      author: selectedVideo.author,
       errorMessage: false,
-      agreedTOC: false // TODO
+      agreedTOC: false // TODO,
     }
     this.onSubmit = this.onSubmit.bind(this)
-    // this.handlePublish = this.handlePublish.bind(this)
-    // this.handlePublish = this.handlePublish.bind(this)
   }
 
   onSubmit (event: Object) {
     event.preventDefault()
 
     paratii.eth.tcr
       .checkEligiblityAndApply(
-        this.props.videoId.toString(),
+        this.props.videoId.id.toString(),
         paratii.eth.web3.utils.toWei(5 + '')
       )
       .then(resp => {
         if (resp && resp === true) {
           this.setState({
             errorMessage: false
           })
-          this.props.onSuccess()
+          const videoToSave = {
+            id: this.state.id,
+            title: this.state.title,
+            description: this.state.description,
+            author: this.state.author,
+            published: true
+          }
+          this.props.saveVideoInfo(videoToSave)
+          this.props.closeModal()
           console.log(
-            `video ${this.props.videoId.toString()} successfully applied to TCR Listing`
+            `video ${this.props.videoId.id.toString()} successfully applied to TCR Listing`
           )
```

Comment:
All of this should really be in an `*Actions.js` module. We don't want this kind of complex logic in components. At the very least there should be a TODO comment added noting this

Reply:
Let's move to a `*Action.js` ?